packages/wdio-sync/src/index.js

```diff
@@ -15,33 +15,31 @@ const log = logger('@wdio/sync')
  * execute test or hook synchronously
  * @param  {Function} fn         spec or hook method
  * @param  {Number}   repeatTest number of retries
+ * @param  {Array}    args       number of retries
  * @return {Promise}             that gets resolved once test/hook is done or was retried enough
  */
-const executeSync = function (fn, repeatTest = 0, args = []) {
-    /**
-     * if a new hook gets executed we can assume that all commands should have finised
-     * with exception of timeouts where `commandIsRunning` will never be reset but here
-     */
-    return new Promise((resolve, reject) => {
-        try {
-            const res = fn.apply(this, args)
-            resolve(res)
-        } catch (e) {
-            if (repeatTest) {
-                return resolve(executeSync(fn, --repeatTest, args))
-            }
-
-            /**
-             * no need to modify stack if no stack available
-             */
-            if (!e.stack) {
-                return reject(e)
-            }
+const executeSync = async function (fn, repeatTest = 0, args = []) {
+    try {
+        let res = fn.apply(this, args)
```

Comment:
@christian-bromann may you please help a bit with validating change in executeSync? 
In some cases fn.apply returns promise and have to be awaited before proceeded further.
This helps me with running before after step hooks and also fixing issue when wdio/sync and async functions are used together. 

There is an alternative way to handle before after step hooks by wrapping them with future, waiting for it, and then catching error ”cannot run without a fiber” but I don't really like that.